compiler/rustc_target/src/spec/abi.rs

```diff
@@ -8,7 +8,7 @@ use rustc_span::{Span, Symbol};
 mod tests;
 
 #[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Copy, Debug)]
-#[derive(HashStable_Generic, Encodable, Decodable)]
+#[derive(HashStable_Generic, Encodable, Decodable, TypeFoldable, TypeVisitable)]
```

Comment:
why is that needed?

Reply:
As with `Unsafety`, this exists because `Abi` also impls `Relate` (used in `<FnSig as Relate>::relate`).

Reply:
Added explanation in 8834fbf4346d25f3051044979379897299ca5e7f